### PR TITLE
:book: Add flake tracking documentation to the user-guide

### PR DESCRIPTION
**Pull request by Rozzii**

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]
LGTM!
A good start. There will probably be things we want to change later if it becomes too large for example, but let's start with this.

File: docs/user-guide/src/flake/introduction.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Introduction to "Flakes"
+
+The Metal3 community maintains a number of tests and Continous Integration pipelines
+and other infrastructure elements in order to be able to test,release,package and
+manage the developement processes of the Metal3 components.
+
```
Comment (fmuyassarov) [inline docs/user-guide/src/flake/introduction.md:6]:
```suggestion
The Metal3 community maintains a number of tests and Continuous Integration pipelines
and other infrastructure elements in order to be able to test, release, package and
manage the development processes of the Metal3 components.

```

File: docs/user-guide/src/flake/introduction.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+In the Metal3 community's nomenclature the word "flake" refers to a recurring error  that appaears
+from time to time in the CI or during or test process but disapears by itself thus making the reproduction
+difficult. In order to eventually figure out and fix the root cause of the flakes the community
+decided to catalog them thus making all developers aware of their existence.
```
Comment (fmuyassarov) [inline docs/user-guide/src/flake/introduction.md:10]:
```suggestion
In the Metal3 community's nomenclature the word "flake" refers to a recurring error  that appears
from time to time in the CI or during or test process but disappears by itself thus making the reproduction
difficult. In order to eventually figure out and fix the root cause of the flakes the community
decided to catalog them thus making all developers aware of their existence.
```

**Comment by fmuyassarov** [comment]
Some types but otherwise lgtm
/approve

**Review by smoshiur1237 — APPROVED** [review]
Thanks for adding the flack tracking docs. Looks good to me.
/lgtm

**Comment by Rozzii** [comment]
Please check if it can be approved @kashifest  @furkatgofurov7

**Pull request merged**